### PR TITLE
backend-common: implement UrlReader.search for the other providers too

### PR DESCRIPTION
**Pull request by freben**

### CODE REVIEW

**Comment by backstage-service** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr4435-pix9c1k2h28xwvh9juyybres5dveoc9v.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/60224aa9353bbb42c9a34d56

**Comment by backstage-service** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr4435-pix9c1k2h28xwvh9juyybres5dveoc9v.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/60224aa9353bbb42c9a34d56

**Comment by backstage-service** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr4435-pix9c1k2h28xwvh9juyybres5dveoc9v.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/60224aa9353bbb42c9a34d56

**Comment by backstage-service** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr4435-pix9c1k2h28xwvh9juyybres5dveoc9v.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/60224aa9353bbb42c9a34d56

**Comment by backstage-service** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr4435-pix9c1k2h28xwvh9juyybres5dveoc9v.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/60224aa9353bbb42c9a34d56

File: packages/backend-common/src/reading/tree/util.ts [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+// Matches a directory name + one `/` at the start of any string,
+// containing any character except `/` one or more times, and ending with a `/`
+// e.g. Will match `dirA/` in `dirA/dirB/file.ext`
+const directoryNameRegex = /^[^\/]+\//;
+
+// Removes the first segment of a forward-slash-separated path
+export function stripFirstDirectoryFromPath(path: string): string {
+  return path.replace(directoryNameRegex, '');
+}
```
Comment (OrkoHunter) [inline packages/backend-common/src/reading/tree/util.ts:25]: Nice to make this shared across both Tar and Zip functions!

**Review by OrkoHunter — APPROVED** [review]

**Comment by OrkoHunter** [comment]
To me, the failing tests looks wrong in the first place since resolving `c/d` at a path will assume that `c/d` is inside the folder of the path in base URL. (Also looking at the new helpers.test.ts inside integration)

https://github.com/backstage/backstage/blob/e57704154842092118a5b84cccd00a1694e90973/plugins/catalog-backend/src/ingestion/processors/LocationEntityProcessor.test.ts#L44-L47

https://github.com/backstage/backstage/blob/e57704154842092118a5b84cccd00a1694e90973/plugins/catalog-backend/src/ingestion/processors/LocationEntityProcessor.test.ts#L53-L55

I wonder why it passed previously. 🤔

**Comment by backstage-service** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr4435-pix9c1k2h28xwvh9juyybres5dveoc9v.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/60224aa9353bbb42c9a34d56

**Comment by freben** [comment]
@OrkoHunter Yeah that was a new behavior that I introduced in the new code - I think it was misguided and chose to let the old test "win" - so it's changed now to suit the old behavior instead.

**Comment by OrkoHunter** [comment]
Awesome! All green ✅ now.

**Comment by backstage-service** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr4435-pix9c1k2h28xwvh9juyybres5dveoc9v.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/60224aa9353bbb42c9a34d56

**Review by benjdlambert — APPROVED** [review]

**Pull request merged**